compiler/damlc/daml-opts/daml-opts/DA/Daml/Options.hs

```diff
@@ -471,7 +471,7 @@ locateCppPath = do
         -- the resources directory because the target produces a tarball which
         -- has the executable directly under the top directory.
         -- See @bazel_tools/packaging/packaging.bzl@.
-      , runfilesPathPrefix = "stackage" </> "hpp-0.6.4" </> "_install" </> "bin"
+      , runfilesPathPrefix = "stackage" </> "hpp-0.6.5" </> "_install" </> "bin"
```

Comment:
So, this was the root cause of the missing field `Settings.sPgm_P` errors: the version bump meant the `hpp` bin moved, so we couldn't locate it, so `adjustDynFlags` couldn't set `sPgm_P` and `sOpt_P`, so the fields were indeed missing so GHC crashed in `doCpp`.

This is not the final fix since we want to prevent this from happening again. I want to remove the hardcoded version of `hpp` here - using a copy instead of the `@stackage/hpp` bin directly should work. I also want to make this crash with a clear error if the file is missing instead of returning a `Nothing` that will silently give us a partial `Settings`.

Reply:
https://github.com/digital-asset/daml/pull/13128/files#r817718726 (!)